src/goto-instrument/contracts/cfg_info.h

```diff
@@ -31,6 +31,51 @@ Date: June 2022
 #include <set>
 #include <vector>
 
+/// Work around the fact that dirtyt does not look into
+/// OTHER instructions
+class dirty_altt : public dirtyt
+{
+public:
+  dirty_altt() : dirtyt()
+  {
+  }
+
+  explicit dirty_altt(const goto_functiont &goto_function)
+    : dirtyt(goto_function)
+  {
+    collect_other(goto_function);
+  }
+
+  explicit dirty_altt(const goto_functionst &goto_functions)
+    : dirtyt(goto_functions)
+  {
+    for(const auto &gf_entry : goto_functions.function_map)
+      collect_other(gf_entry.second);
+  }
+
+protected:
+  void collect_other(const goto_functiont &goto_function)
+  {
+    for(const auto &i : goto_function.body.instructions)
+    {
+      if(i.is_other())
+      {
+        auto &statement = i.get_other().get_statement();
+        if(
+          statement == ID_array_set || statement == ID_array_copy ||
+          statement == ID_array_replace || statement == ID_array_equal ||
+          statement == ID_havoc_object)
+        {
+          for(const auto &op : i.get_other().operands())
+          {
+            find_dirty(op);
+          }
+        }
+      }
+    }
+  }
```

Comment:
LGTM, but why not incorporate this change into `dirtyt` class itself?

Reply:
Modifying `dirtyt` could have wide-ranging repercussions since many other analyses depend on it.